src/v/coproc/script_context.cc

```diff
@@ -65,29 +69,49 @@ script_context::script_context(
       "valid subscription list");
 }
 
+static void exc_printer(const std::exception_ptr& eptr) {
+    try {
+        /// If an exception is unhandled by this method, it will percolate to
+        /// the caller of start(). ATM by design only one exception should do
+        /// this, the \ref script_failed_exception.
+        std::rethrow_exception(eptr);
+    } catch (const std::logic_error& e) {
+        vlog(
+          coproclog.error,
+          "std::logic_error observed wasm engine sent unexpected reply, "
+          "default behavior is to retry, exception: {}",
+          e.what());
```

Comment:
libfmt knows what to do with exceptions, you can pass just `e` (also twice more below).